core-bundle/contao/library/Contao/Pagination.php

```diff
@@ -147,9 +147,11 @@ public function __construct($intRows, $intPerPage, $intNumberOfLinks=7, $strPara
 		$this->lblLast = $GLOBALS['TL_LANG']['MSC']['last'];
 		$this->lblTotal = $GLOBALS['TL_LANG']['MSC']['totalPages'];
 
-		if (Input::get($strParameter) > 0)
+		$input = System::getContainer()->get('contao.framework')->getAdapter(Input::class);
+
+		if ($input->get($strParameter) > 0)
 		{
-			$this->intPage = (int) Input::get($strParameter);
+			$this->intPage = (int) $input->get($strParameter);
```

Comment:
These changes seem unrelated to the PR. And besides, you do not need to fetch an adapter here anyway, since there are no tests.

Reply:
Oh 🙈